### PR TITLE
Release workflow improvements

### PR DESCRIPTION
**Pull request by theetrain**
Introducing some semi-automatic release tooling:

- remove `husky`
- add linting to PR pipelines
- add release guide to contributing.md, and table of contents
- add `yarn release` semi-automatic script, leveraging the `standard-version` package
- Lint and fix neglected files. I assume their line endings were incorrect

How to test:

1. Check guide in CONTRIBUTING.md
2. Run `yarn release`. This won't create a commit nor tag; those steps are manual. Note the generated changelog, package.json version bump, and documentation version bump

Next steps: once we're comfortable with these tools, we'll set up continuous delivery.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (metonym) [inline CONTRIBUTING.md:163]: Can we add a note to this section that it applies to maintainers only?

File: package.json
Comment (metonym) [inline package.json:76]: I'd prefer to preserve existing users on this list. `josefaidt` was a maintainer at IBM.

**Review by metonym — COMMENTED** [review]
Thanks for setting this up.

Tested locally and works as expected. This LGTM pending a couple of comments.

**Comment by theetrain** [comment]
Thank you; all comments were addressed.

**Review by metonym — APPROVED** [review]

**Comment by metonym** [comment]
This is great, especially moving the lint to CI since it caused trouble for some users when they tried contributing. Anything to make the contributing flow better and the release flow automated is a big win.

**Review by freezernick — COMMENTED** [review]
Actually `standard-version` is [deprecated](https://github.com/conventional-changelog/standard-version/issues/919) and the recommended successor is https://github.com/absolute-version/commit-and-tag-version

The only changes required are to replace every `standard-version` with `commit-and-tag-version` ( and update the dependencies )

**Pull request merged**
